dashboard/test/ui/test_status.js

```diff
@@ -91,8 +93,13 @@ Test.prototype.fetchStatus = function () {
     this.updateView();
   };
 
-  fetch(`${API_BASEPATH}/${this.s3Key()}`, {mode: 'no-cors'})
-    .then(response => response.json())
+  return fetch(`${API_BASEPATH}/${this.s3Key()}`, {mode: 'no-cors'})
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error(`While fetching status, "${response.url}" returned ${response.status}.`);
```

Comment:
Here we have the appropriate way to check whether a fetch request is successful, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful).